Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -688,10 +692,26 @@ void addRecursiveQueries(
       std::map<std::string, ROMOL_SPTR>::const_iterator iter =
           queries.find(pval);
       if (iter == queries.end()) {
+        notFound = true;
+      }
+      else {
+        qToAdd = new RecursiveStructureQuery(new ROMol(*(iter->second)));
+      }
+    }
+
+    if (notFound) {
+      // See if we are actually a smarts expression already
+      RWMol *m = 0;
+      try {
+        m = SmartsToMol(maybeSmarts);
+        if (!m)
+          throw KeyErrorException(pval);
```

Comment:
Doesn't seem like this should be a `KeyErrorException`
Maybe a `ValueErrorException`?

Reply:
I think it's still a KeyError exception, in this case it's a double lookup:

```
a.SetProp("foo", "patternname")
```

and we are allowing
```
a.SetProp("foo", "[C&R6]")
```

We don't know if it is a key lookup failure or a smarts failure as most keys will also not be parseable as smarts.